cpu/efm32/periph/spi.c

```diff
@@ -56,6 +59,69 @@ void spi_init_pins(spi_t bus)
     gpio_init(spi_config[bus].miso_pin, GPIO_IN_PD);
 }
 
+spi_clk_t spi_get_clk(spi_t bus, uint32_t freq)
+{
+    (void)bus;
+
+    /* bound divider to 65536 */
+    if (freq < DIV_ROUND_UP(CLOCK_CORECLOCK, 65536)) {
+        return (spi_clk_t){ .err = -EDOM };
+    }
+
+    /* =================================================================
+     * from USART_BaudrateSyncSet() of the gecko SDK
+     */
+    uint32_t ref_freq, clkdiv;
+
+    /* Prevent dividing by 0. */
+    EFM_ASSERT(freq);
+
+    /*
+     * CLKDIV in synchronous mode is given by:
+     *
+     * CLKDIV = 256 * (fHFPERCLK/(2 * br) - 1)
+     */
+
+    /* HFPERCLK/HFPERBCLK used to clock all USART/UART peripheral modules. */
+#if defined(_SILICON_LABS_32B_SERIES_2)
+    ref_freq = CMU_ClockFreqGet(cmuClock_PCLK);
+#else
+#if defined(_CMU_HFPERPRESCB_MASK)
+    if (dev == USART2) {
+        ref_freq = CMU_ClockFreqGet(cmuClock_HFPERB);
+    }
+    else {
+        ref_freq = CMU_ClockFreqGet(cmuClock_HFPER);
+    }
+#else
+    ref_freq = CMU_ClockFreqGet(cmuClock_HFPER);
+#endif
+#endif
+
+    clkdiv = (ref_freq - 1) / (2 * freq);
+    clkdiv = clkdiv << 8;
+
+    /* Verify that resulting clock divider is within limits. */
+    EFM_ASSERT(!(clkdiv & ~_USART_CLKDIV_MASK));
```

Comment:
Hmmm, since a user could theoretically set the frequency, can we avoid the assert and truncate or something?  Since there  can be a difference between the set frequency and the real frequency there would be a verify step anyways if it is important.

Reply:
I simply removed it, as it would never have been triggered anyway, since the limit is already checked at the beginning of the function.